commands/load_balancers.go

```diff
@@ -557,17 +557,17 @@ func buildRequestFromArgs(c *CmdConfig, r *godo.LoadBalancerRequest) error {
 		r.HTTPIdleTimeoutSeconds = &t
 	}
 
-	allowRules, err := c.Doit.GetStringSlice(c.NS, doctl.ArgAllowList)
+	allowRules, err := c.Doit.GetStringSlicePtr(c.NS, doctl.ArgAllowList)
 	if err != nil {
 		return err
 	}
 
-	denyRules, err := c.Doit.GetStringSlice(c.NS, doctl.ArgDenyList)
+	denyRules, err := c.Doit.GetStringSlicePtr(c.NS, doctl.ArgDenyList)
 	if err != nil {
 		return err
 	}
 
-	if len(allowRules) > 0 || len(denyRules) > 0 {
+	if allowRules != nil || denyRules != nil || len(allowRules) > 0 || len(denyRules) > 0 {
```

Comment:
Nit: length of `nil` will still be `0`. 
```suggestion
	if len(allowRules) > 0 || len(denyRules) > 0 {
```

Reply:
I don't think this logic is quite right either. If the `Firewall` object that is supplied to the API is missing, we will not update the rules. So we need to construct a zero Firewall struct when `allowRules` or `denyRules` has been set but is empty. I think there are three examples when a customer updates a firewall:

1) Neither `--allow-list` or `--deny-list` is provided, in this case we should have a `nil` Firewall which would ensure that we don't update the firewall.
2) `--allow-list` or `--deny-list` is provided but no values are provided. In this case, we should send along a zero Firewall object`
3) `--allow-list` or `--deny-list` is provided with values and we should pass a Firewall object with them provided.

I think `GetStringSlicePtr` will need to return a bool to indicate that the param was provided on the CLI call.

Reply:
> 1. Neither --allow-list or --deny-list is provided, in this case we should have a nil Firewall which would ensure that we don't update the firewall.

If the `--allow-list `or `--deny-list` is not provided the `GetStringSlicePtr` should return nil which should ignore updating the firewall.

> 2. --allow-list or --deny-list is provided but no values are provided. In this case, we should send along a zero Firewall object`

If the `--allow-list `or `--deny-list` is provided but empty the `GetStringSlicePtr` should return an empty slice that is defined which will pass the `!= nil` check causing the firewall to be updated

> 3. --allow-list or --deny-list is provided with values and we should pass a Firewall object with them provided.

If the `--allow-list `or `--deny-list` is provided the `GetStringSlicePtr` should return a populated slice that is defined which will pass the `!= nil` check causing the firewall to be updated

The original GetStringSlice always returns at a minimum a defined but empty slice. The GetStringSlicePtr allows a nil slice to return when the flag is not provided. This allows us to differentiate between a defined by empty slice and a nil slice. I might be missing something.